src/group_theory/coset.lean

```diff
@@ -3,8 +3,37 @@ Copyright (c) 2018 Mitchell Rowett. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Mitchell Rowett, Scott Morrison
 -/
+
 import group_theory.subgroup
 
+/-!
+# Cosets
+
+This file develops the basic theory of left and right cosets.
+
+## Main definitions
+
+* `left_coset a s`: the left coset `a * s` corresponding to an element `a : α` and a subset `s ⊆ α`.
+* `right_coset s a`: the right coset `s * a` corresponding to an element `a : α` and a subset
+  `s ⊆ α`.
+* `quotient s`: the quotient type corresponding to the left cosets with respect to a subgroup `s`.
+* `mk`: the canonical map from `α` to `α/s` for a subgroup `s` of `α`.
+* `left_coset_equiv_subgroup`: the natural bijection between a left coset and the subgroup.
+
+## Notation
+
+* `a *l s`: for a `has_mul α` , i.e. a magma, `a : α` and `s ⊆ α`, the left coset `a * s`.
```

Comment:
```suggestion
* `a *l s`: for the left coset `a * s`.
```
Don't think all of the details about the types of all of the arguments are necessary to be honest. (Same for the next few lines)

Reply:
I changed it to link to `left_coset a s` without spelling out the details. Thanks for the suggestion.